### ISSUE TITLE
Race condition building site from a clean state when there is a page with a `post-list` directive.

### ISSUE THREAD

**Issue by chaghi** [getnikola/nikola#1672]
When a site has a page with a `post-list` directive, (re)building the site from a clean state generates a `FileNotFoundError` exception similar to this:
```console
(nikola)mariano@arya:~/Devel/test$ nikola build
Scanning posts....done!
.  render_listings:output/listings/index.html
.  render_sources:output/stories/first-page.rst
.  render_sources:output/posts/first-post.rst
.  render_tags:output/categories/index.html
.  render_tags:output/assets/js/tag_cloud_data.json
.  render_posts:cache/stories/first-page.html
Traceback (most recent call last):
  File "/home/mariano/.virtualenvs/nikola/lib/python3.4/site-packages/doit/doit_cmd.py", line 121, in run
    return command.parse_execute(args)
  File "/home/mariano/.virtualenvs/nikola/lib/python3.4/site-packages/doit/cmd_base.py", line 84, in parse_execute
    return self.execute(params, args)
  File "/home/mariano/.virtualenvs/nikola/lib/python3.4/site-packages/doit/cmd_base.py", line 278, in execute
    return self._execute(**exec_params)
  File "/home/mariano/.virtualenvs/nikola/lib/python3.4/site-packages/doit/cmd_run.py", line 215, in _execute
    return runner.run_all(self.control.task_dispatcher())
  File "/home/mariano/.virtualenvs/nikola/lib/python3.4/site-packages/doit/runner.py", line 237, in run_all
    self.run_tasks(task_dispatcher)
  File "/home/mariano/.virtualenvs/nikola/lib/python3.4/site-packages/doit/runner.py", line 204, in run_tasks
    self.process_task_result(node, catched_excp)
  File "/home/mariano/.virtualenvs/nikola/lib/python3.4/site-packages/doit/runner.py", line 175, in process_task_result
    self.dep_manager.save_success(task)
  File "/home/mariano/.virtualenvs/nikola/lib/python3.4/site-packages/doit/dependency.py", line 413, in save_success
    timestamp = os.path.getmtime(dep)
  File "/home/mariano/.virtualenvs/nikola/lib/python3.4/genericpath.py", line 55, in getmtime
    return os.stat(filename).st_mtime
FileNotFoundError: [Errno 2] No such file or directory: 'cache/posts/first-post.html'
```

*Steps to reproduce*

* `nikola init test` (accept all the defaults)
* add a post
* add a page with a `post-list` directive
* `nikola build`

*Expected result*

* The site should be built for the first time.

*Actual result*

* `FileNotFoundError` exception.

*Remarks*

* The error is always triggered when building from a clean state, i.e., first build, or first build after a `clean`/`forget`, or a manual delete of `cache/`, `output/` and `.doit.db`.
* Subsequent builds after the error run ok, until the state is cleared again.

*Known workaround*

Setting the date of the page with the `post-list` to a past value (like `1970-01-01 00:00:00`) avoids the race condition. This was suggested in the mailing-list: https://groups.google.com/d/msg/nikola-discuss/ULhZ3mIG0yk/3740m_9ItLYJ

*Environment*

Ubuntu 14.04, Python 3.4.0, Nikola 7.3.1.1, with the following packages:
```console
blinker (1.3)
certifi (14.5.14)
docutils (0.12)
doit (0.27.0)
ghp-import (0.4.1)
ipython (3.1.0)
Jinja2 (2.7.3)
jsonschema (2.4.0)
livereload (2.3.2)
Logbook (0.9.0)
lxml (3.4.3)
Mako (1.0.1)
Markdown (2.6.2)
MarkupSafe (0.23)
micawber (0.3.2)
mistune (0.5.1)
natsort (3.5.6)
Nikola (7.3.1.1)
phpserialize (1.3)
Pillow (2.8.1)
pip (6.1.1)
ptyprocess (0.4)
pygal (1.7.0)
Pygments (2.0.2)
pyinotify (0.9.5)
Pyphen (0.9.1)
PyRSS2Gen (1.1)
python-dateutil (2.4.2)
pyzmq (14.6.0)
requests (2.6.2)
setuptools (15.1)
six (1.9.0)
smartypants (1.8.6)
terminado (0.5)
tornado (4.1)
typogrify (2.0.7)
Unidecode (0.4.17)
webassets (0.10.1)
Yapsy (1.11.23)
```

**Comment by ralsina** [comment]
This would indicate we are not using the docutils dependency tracker correctly. That page should depend on 'cache/posts/first-post.html' which would make everything build in the right order.

Right now it does this:

```
            bp = post.translated_base_path(lang)
            if os.path.exists(bp):
                self.state.document.settings.record_dependencies.add(bp)
```

Which doesn't make much sense at first sight since we should depend on the file whether it exists or not :-)

**Comment by ralsina** [comment]
I *think* this shoud be fixed by #1673 too

**Comment by chaghi** [comment]
Yes, it's fixed. I couldn't reproduce this on the `fix-1671` branch.

Thank you!
